Password is incorrect
'password'
'password'
'password'
'password' is required
'password' is required
password is required
username is required
User does not exist
500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
403 Forbidden: <Permission needs={Need(method='role', value='admin')} excludes=set()>
403 Forbidden: <Permission needs={Need(method='role', value='admin')} excludes=set()>
403 Forbidden: <Permission needs={Need(method='role', value='admin')} excludes=set()>
403 Forbidden: <Permission needs={Need(method='role', value='admin')} excludes=set()>
403 Forbidden: <Permission needs={Need(method='role', value='admin')} excludes=set()>
403 Forbidden: <Permission needs={Need(method='role', value='admin')} excludes=set()>
403 Forbidden: <Permission needs={Need(method='role', value='admin')} excludes=set()>
Only admins can login here.
Only admins can login here.
Only users can login here.
Password is incorrect
username is required
{'password': ['Password must contain at least one uppercase character']}
{'username': ['Username must be at least 6 characters long']}
Role does not exist
(sqlite3.OperationalError) no such column: product.added_by
[SQL: SELECT product.id AS product_id, product.product_name AS product_product_name, product.product_rate AS product_product_rate, product.product_unit AS product_product_unit, product.product_description AS product_product_description, product.current_stock AS product_current_stock, product.expiry_date AS product_expiry_date, product.category_id AS product_category_id, product.added_by AS product_added_by 
FROM product]
(Background on this error at: https://sqlalche.me/e/20/e3q8)
Entity namespace for "role" has no property "name"
{'role_id': ['Unknown field.'], 'password': ['Password must contain at least one uppercase character']}
{'role_id': ['Unknown field.']}
{'role_id': ['Unknown field.']}
{'role_id': ['Unknown field.']}
UserSchema.make_user() got an unexpected keyword argument 'many'
UserSchema.make_user() got an unexpected keyword argument 'many'
UserSchema.make_user() got an unexpected keyword argument 'many'
UserSchema.make_user() got an unexpected keyword argument 'many'
'many' is an invalid keyword argument for print()
Schema.dumps() missing 1 required positional argument: 'obj'
(sqlite3.IntegrityError) UNIQUE constraint failed: user.email
[SQL: INSERT INTO user (username, password, email, role_id) VALUES (?, ?, ?, ?)]
[parameters: ('manager2', 'scrypt:32768:8:1$s3DI6VeAT9pP3wve$d4e39852eaa783110d2a89e38f43dc505fb9eb49d831b3b641cc4c5a154d906f8e7158603709c1f935166779ecd8e4a9352c0bc0fa505f2a260b2998f60f92dd', 'manager2@grocery.com', 3)]
(Background on this error at: https://sqlalche.me/e/20/gkpj)
(sqlite3.IntegrityError) UNIQUE constraint failed: user.username
[SQL: INSERT INTO user (username, password, email, role_id) VALUES (?, ?, ?, ?)]
[parameters: ('manager2', 'scrypt:32768:8:1$nrT4TQJ1JQ2jDiDe$a54f0c00fa568e5729652e2293ea0a8322c4f6b14d25652797aea39c19fdae7941c76eb6623c2b2cf8e8f2cd4579c98cde1bb0783a687a18627bd6ec9517a20c', 'manager3@grocery.com', 3)]
(Background on this error at: https://sqlalche.me/e/20/gkpj)
(sqlite3.IntegrityError) UNIQUE constraint failed: user.username
[SQL: INSERT INTO user (username, password, email, role_id) VALUES (?, ?, ?, ?)]
[parameters: ('manager2', 'scrypt:32768:8:1$J9H9019RxjT1QkFo$acd283be767a18d1e85a9bc750ae793ed9b3a35fc1372e0dd90299d779943f57dc72c53b9389b8b25cb0b41dfcb4afc414a7f6d0e4aa8681689fb628ffd74226', 'manager3@grocery.com', 3)]
(Background on this error at: https://sqlalche.me/e/20/gkpj)
{'current_stock': ['Unknown field.'], 'unit': ['Unknown field.'], 'name': ['Unknown field.'], 'price': ['Unknown field.'], 'description': ['Unknown field.']}
{'current_stock': ['Unknown field.'], 'price': ['Unknown field.'], 'name': ['Unknown field.'], 'unit': ['Unknown field.'], 'description': ['Unknown field.']}
{'unit': ['Unknown field.'], 'name': ['Unknown field.'], 'price': ['Unknown field.'], 'description': ['Unknown field.'], 'current_stock': ['Unknown field.']}
{'rate': ['Unknown field.'], 'unit': ['Unknown field.'], 'description': ['Unknown field.'], 'name': ['Unknown field.'], 'current_stock': ['Unknown field.']}
{'unit': ['Unknown field.'], 'current_stock': ['Unknown field.'], 'rate': ['Unknown field.'], 'name': ['Unknown field.'], 'description': ['Unknown field.']}
the JSON object must be str, bytes or bytearray, not dict
{'unit': ['Unknown field.'], 'rate': ['Unknown field.'], 'name': ['Unknown field.'], 'description': ['Unknown field.']}
ProductSchema.make_product() got an unexpected keyword argument 'many'
{'_schema': ["__init__() got an unexpected keyword argument 'added_by'"]}
{'_schema': ["__init__() got an unexpected keyword argument 'category_id'"]}
{'_schema': ["__init__() missing 4 required positional arguments: 'name', 'rate', 'unit', and 'description'"]}
"product_description" field does not exist.
(sqlite3.OperationalError) table product has no column named name
[SQL: INSERT INTO product (name, rate, unit, description, current_stock, expiry_date, category_id, added_by) VALUES (?, ?, ?, ?, ?, ?, ?, ?)]
[parameters: ('apple', 1.99, 'kg', 'red apple from china', 100, '2023-12-31', 1, 3)]
(Background on this error at: https://sqlalche.me/e/20/e3q8)
Object of type Response is not JSON serializable
{'email': ['Email already exists'], 'username': ['Username already exists']}
Object of type Response is not JSON serializable
Invalid fields for <CategorySchema(many=False)>: OrderedSet(['products']).
(sqlite3.IntegrityError) UNIQUE constraint failed: product.name
[SQL: INSERT INTO product (name, rate, unit, description, current_stock, expiry_date, category_id, added_by) VALUES (?, ?, ?, ?, ?, ?, ?, ?)]
[parameters: ('apple', 1.99, 'kg', 'red apple from china', 100, '2023-12-31', 1, 2)]
(Background on this error at: https://sqlalche.me/e/20/gkpj)
{'name': ['Product with name apple already exists']}
Invalid fields for <CategorySchema(many=False)>: OrderedSet(['products']).
{'name': ['Product with name apples already exists']}
Schema.dumps() missing 1 required positional argument: 'obj'
Object of type Category is not JSON serializable
Invalid fields for <CategorySchema(many=False)>: OrderedSet(['products']).
Object of type Response is not JSON serializable
Object of type Response is not JSON serializable
